twill-yarn/src/main/java/org/apache/twill/yarn/YarnTwillController.java

```diff
@@ -77,7 +77,7 @@
    */
   YarnTwillController(String appName, RunId runId, ZKClient zkClient,
                       final ApplicationMasterLiveNodeData amLiveNodeData, final YarnAppClient yarnAppClient) {
-    super(runId, zkClient, Collections.<LogHandler>emptyList());
+    super(appName, runId, zkClient, amLiveNodeData.getKafkaZKConnect() != null, Collections.<LogHandler>emptyList());
```

Comment:
Yeah, thinking about having `ApplicationMasterLiveNode.isLogCollectionEnabled` to make the contract more clear. Otherwise later we will forget why checking for getKafkaZKConnect != null here